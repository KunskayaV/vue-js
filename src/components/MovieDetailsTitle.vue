<template>
  <div class="title-container">
    <div class="title-wrapper">
      <span class="movie-title" date-testid="details-title">{{ props.title }}</span>
      <span class="movie-rating" date-testid="details-rating">{{ props.rating }}</span>
    </div>
    <span class="movie-genre" date-testid="details-genre">{{ props.genre }}</span>
  </div>
</template>

<script setup lang="ts">
const props = defineProps<{ title: string; rating: number; genre: string }>()
</script>

<style scoped>
.title-container {
  display: flex;
  flex-direction: column;
}

.title-wrapper {
  display: flex;
  align-items: center;
  gap: 12px;
}

.movie-title {
  font-size: 4rem;
  line-height: 5rem;
  color: #ffffff;
}

.movie-rating {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 60px;
  border: 1px solid #ffffff;
  border-radius: 50%;
  font-size: 1.2rem;
  color: #a1e66f;
}

.movie-genre {
  margin-top: 8px;
  font-size: 0.9rem;
  color: #ffffff;
}
</style>
