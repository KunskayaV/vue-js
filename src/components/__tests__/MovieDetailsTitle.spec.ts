import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import MovieDetailsTitle from '../MovieDetailsTitle.vue'

describe('MovieDetailsTitle', () => {
  const props = {
    title: 'Title',
    rating: 4.2,
    genre: 'Drama'
  }
  it('should match snapshot', () => {
    const wrapper = mount(MovieDetailsTitle, { props })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correct data', () => {
    const wrapper = mount(MovieDetailsTitle, { props })

    expect(wrapper.find('[date-testid=details-title]').text()).toBe(props.title)
    expect(wrapper.find('[date-testid=details-rating]').text()).toBe(String(props.rating))
    expect(wrapper.find('[date-testid=details-genre]').text()).toBe(props.genre)
  })
})
