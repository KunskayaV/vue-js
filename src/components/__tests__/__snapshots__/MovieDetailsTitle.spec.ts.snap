// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieDetailsTitle > should match snapshot 1`] = `
"<div data-v-8cdb4aee=\\"\\" class=\\"title-container\\">
  <div data-v-8cdb4aee=\\"\\" class=\\"title-wrapper\\"><span data-v-8cdb4aee=\\"\\" class=\\"movie-title\\" date-testid=\\"details-title\\">Title</span><span data-v-8cdb4aee=\\"\\" class=\\"movie-rating\\" date-testid=\\"details-rating\\">4.2</span></div><span data-v-8cdb4aee=\\"\\" class=\\"movie-genre\\" date-testid=\\"details-genre\\">Drama</span>
</div>"
`;
